packages/teams-js/src/internal/communication.ts

```diff
@@ -325,8 +324,11 @@ function shouldProcessMessage(messageSource: Window, messageOrigin: string): boo
     messageOrigin === Communication.currentWindow.location.origin
   ) {
     return true;
+  } else if (Communication.parentOrigin && messageOrigin && messageOrigin === Communication.parentOrigin) {
```

Comment:
>Communication.parentOrigin && messageOrigin && messageOrigin === Communication.parentOrigin [](http://example.com/codeflow?start=13&length=91)

What is this new check doing / why is it doing it now?

The code in this file is overall pretty hard to follow and this function in particular can be tricky to understand. One thing that I would encourage to help make it a little easier to read is to consider extracting this new condition into a short one-line boolean function that is named with a clear name.

I don't know what this is doing so I'll make something up just to illustrate my point. Compare:
```
  } else if (Communication.parentOrigin && messageOrigin && messageOrigin === Communication.parentOrigin) {
    return true;
```
to
```
  } else if (isMessageComingFromParentOrigin()) {
    return true;
```

Again I don't know what this new check is checking but a function name that describes the check could help.